x/auth/tx/module/module.go

```diff
@@ -0,0 +1,109 @@
+package tx
+
+import (
+	"fmt"
+
+	modulev1 "cosmossdk.io/api/cosmos/tx/module/v1"
+	"cosmossdk.io/core/appmodule"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/depinject"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/cosmos/cosmos-sdk/x/auth/posthandler"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	feegrantkeeper "github.com/cosmos/cosmos-sdk/x/feegrant/keeper"
+)
+
+func init() {
+	appmodule.Register(&modulev1.Module{},
+		appmodule.Provide(provideModule),
+	)
+}
+
+type txInputs struct {
+	depinject.In
+
+	Config              *modulev1.Module
+	ProtoCodecMarshaler codec.ProtoCodecMarshaler
+
+	AccountKeeper  *authkeeper.AccountKeeper `key:"cosmos.auth.v1.AccountKeeper" optional:"true"`
+	BankKeeper     bankkeeper.Keeper         `key:"cosmos.bank.v1.Keeper" optional:"true"`
+	FeeGrantKeeper *feegrantkeeper.Keeper    `key:"cosmos.feegrant.v1.Keeper" optional:"true"`
```

Comment:
Should we use expected keeper interfaces here?

Reply:
Ideally yes

Reply:
Changed!